src/Shared/PSBTSign.razor

```diff
@@ -13,54 +15,69 @@
+                <MemoEdit Rows="4" @bind-Text="@TemplatePsbtString" Disabled="true" ElementId="psbt-to-sign"></MemoEdit>
             </Alert>
             <Fields Flex="Flex.JustifyContent.Center">
                 <Field>
-                    <Validation Validator="ValidatePSBT">
+                    <Validation @ref="ValidatePSBTValidator" Validator="PSBTValidator">
                         <FieldLabel>To sign this operation, please paste here your PSBT in base64 string</FieldLabel>
-                        <MemoEdit Rows="4" @bind-Text="@SignedPSBT">
-                            <ValidationError />
+                        <MemoEdit Rows="4" @ref="_PSBTField" @onchange="HandleOnChange" @bind-Text="@SignedPSBT" ElementId="psbt-to-paste">
+                            <ValidationError/>
                         </MemoEdit>
                     </Validation>
                 </Field>
             </Fields>
+            <Fields Visibility="Visibility.Invisible" Style="height: 0" ElementId="channel-amount">@(ChannelRequest?.Amount ?? WithdrawalRequest?.Amount)</Fields>
+            <Fields Visibility="Visibility.Invisible" Style="height: 0"  ElementId="request-type">@GetRequestType()</Fields>
+            <Field id="approve-with-extension">Or click on your nodeguard signing extension to automatically sign</Field>
         </ModalBody>
         <ModalFooter>
             <Button Color="Color.Secondary" Clicked="@HideModal">Cancel</Button>
-            <Button Color="Color.Primary" Disabled="_isSignedPSBTInvalid" Clicked="@ApproveRequestDelegate">
+            <Button id="approve-button" Color="Color.Primary" Disabled="_isSignedPSBTInvalid" Clicked="@ApproveRequestDelegate">
                 Approve
             </Button>
         </ModalFooter>
     </ModalContent>
 </Modal>
 
+@inject IJSRuntime JsRuntime
 @code {
-
     [Inject]
     public IToastService ToastService { get; set; }
 
     [Parameter]
-    public int? RequestId { get; set; }
+    public ChannelOperationRequest? ChannelRequest { get; set; }
+    [Parameter]
+    public WalletWithdrawalRequest? WithdrawalRequest { get; set; }
 
     [Parameter]
     public string TemplatePsbtString { get; set; }
 
     [Parameter]
     public Action ApproveRequestDelegate { get; set; }
 
-    [Parameter] 
+    [Parameter]
     public  string? SignedPSBT { get; set; }
 
     [Parameter]
     public SigHash SigHashMode { get; set; } = SigHash.All;
 
     private bool _isSignedPSBTInvalid = true;
     private Modal? _modalRef;
+    private MemoEdit _PSBTField;
+    private Validation ValidatePSBTValidator;
+
+    private void HandleOnChange(ChangeEventArgs e)
+    {
+        // HACK: We get the data from the "change" event in the extension and pass it to blazor
+        // otherwise it will not rerender when we put the value inside the textarea
+        SignedPSBT = e.Value?.ToString();
+        ValidatePSBT(SignedPSBT);
+    }
```

Comment:
we might remove this in the future if we make the signer communicate with the nodeguard